core/src/main/scala/cats/data/Chain.scala

```diff
@@ -180,6 +180,36 @@ sealed abstract class Chain[+A] {
     result
   }
 
+  /**
+   * Splits the chain into two, placing the initial `n` elements in the first element of the tuple
+   * and the remaining elements in the second element of the tuple. If `n` is greater than the length
+   * of this Chain, the first element of the tuple will contain the entire Chain and the second element
+   * will be an empty chain.
+   */
+  final def splitAt(n: Int): (Chain[A], Chain[A]) = {
+    @tailrec
+    def loop(togo: Int, acc: Chain[A], rest: Chain[A]): (Chain[A], Chain[A]) = {
+      if (togo == 0) (acc, rest)
+      else
+        rest match {
+          case Empty            => (acc, Empty)
+          case s @ Singleton(_) => (Chain.concat(acc, s), Empty)
+          case Wrap(seq) =>
+            val (l, r) = seq.splitAt(togo)
+            (Chain.concat(acc, Wrap(l)), Wrap(r))
```

Comment:
this Wrap is not safe... I think an invariant of Wrap is that it has length 2 or more. By manually constructing this, you could violate that. I think we should use fromSeq on both of these.

Reply:
Will fix.